lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -444,7 +445,9 @@ std::error_code ImplicitSerializedModuleLoader::findModuleFilesInDirectory(
           (!ModuleBuffer && !ModuleDocBuffer)) &&
          "Module and Module Doc buffer must both be initialized or NULL");
 
-  if (LoadMode == ModuleLoadingMode::OnlyInterface)
+  if (LoadMode == ModuleLoadingMode::OnlyInterface ||
+      (Ctx.IgnoreAdjacentModules && ModuleID.Item.str() != STDLIB_NAME &&
+       ModuleID.Item.str() !=SWIFT_STRING_PROCESSING_NAME))
```

Comment:
Nitpick: space after `!=`. But if there's not any other changes I'd just leave it.